src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -2,22 +2,121 @@
 
 namespace trace {
 
+AssemblyInfo GetAssemblyInfo(ICorProfilerInfo3* info,
+                             const AssemblyID& assembly_id) {
+  std::wstring name(kNameMaxSize, 0);
+  unsigned long name_len = 0;
+  auto hr = info->GetAssemblyInfo(assembly_id, (unsigned long)(name.size()),
+                                  &name_len, name.data(), nullptr, nullptr);
+  if (FAILED(hr) || name_len == 0) {
+    return {};
+  }
+  name = name.substr(0, name_len - 1);
+  return {assembly_id, name};
+}
+
 std::wstring GetAssemblyName(
     const ComPtr<IMetaDataAssemblyImport>& assembly_import,
     const mdAssemblyRef& assembly_ref) {
-  const unsigned long str_max = 512;
-  std::wstring str(str_max, 0);
-  unsigned long str_len = 0;
+  std::wstring name(kNameMaxSize, 0);
+  unsigned long name_len = 0;
   ASSEMBLYMETADATA assembly_metadata{};
-  DWORD assembly_flags = 0;
+  unsigned long assembly_flags = 0;
```

Comment:
Some type defs like `DWORD` should be fine to use since they are defined in the CoreCLR headers which we have to include to use the Profiling API. [Google's C++ style guide](https://google.github.io/styleguide/cppguide.html#Windows_Code) also allows using these (are we following that guide?). However, they do suggest keeping "as close as you can to the underlying C++ types. For example, use `const TCHAR *` instead of `LPCTSTR`."